AngularLanguageService.2022/LanguageServer/DiagnosticFormatterMiddleLayer.cs

```diff
@@ -18,8 +18,23 @@ public DiagnosticFormatterMiddleLayer() : base(methodsToHandle: Methods.TextDocu
 
         public override Task HandleNotificationAsync(string methodName, JToken methodParam, Func<JToken, Task> sendNotification)
         {
-            // TODO: Format diagnostics (see #47).
-            return base.HandleNotificationAsync(methodName, methodParam, sendNotification);
+            var diagnosticParams = methodParam.ToObject<PublishDiagnosticParams>();
+
+            foreach (Diagnostic diagnostic in diagnosticParams.Diagnostics)
+            {
+                // Add an Angular code prefix
+                diagnostic.Code = "(NG) " + diagnostic.Code?.Value;
```

Comment:
it looks like these diagnostics are usually formatted like `ngtsc(-123456)`: https://stackoverflow.com/questions/67022761/angular-path-is-not-a-known-element

Should we go with that format instead?

Reply:
@zkat that seems to be specific to the Angular compiler. I had chosen `NG` since it is the one used in [the error docs](https://angular.io/errors/NG0100), but I don't have very strong opinions about it.

Reply:
I mean, it seems to be how other editors display it: https://stackoverflow.com/questions/72974385/opening-tag-input-not-terminated-ngtsc-995002

This class of errors seems to be distinct from the NG0000 errors on the angular error site, too. I'm not sure where they come from, exactly, but they might be specific to the angular tsc compiler?

Reply:
You're right, VS Code also uses `ngtsc()`. Let's keep that convention!:
![image](https://user-images.githubusercontent.com/62502207/199117557-5f6683b7-fe8b-4529-bd10-42bc0aa32611.png)